### ISSUE TITLE
Podman init/start causes kernel panic on 2021 M1 Macbook Pro 16-inch

### ISSUE THREAD

**Issue by akirilov** [containers/podman#14633]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

**Steps to reproduce the issue:**

1. On an M1 16-inch Macbook Pro (2021):

2. `podman machine init`

3. `podman machine start`

**Describe the results you received:**
Kernel panic (due to unsupported use of qemu `highmem` flag)

**Describe the results you expected:**
No kernet panic

**Additional information you deem important (e.g. issue happens only occasionally):**
Reproduces reliably. Issue is caused by commit 8d3e6577ae4b5c6a6ed25ff5c332df042094e415. Tried the following to verify:

1. Pull latest
2. Build
3. Repro as above. Result: panic
4. Revert 8d3e6577ae4b5c6a6ed25ff5c332df042094e415
5. Build
6. Repro as above. Result: no panic (however, cannot run high mem VMs)

This is a known issue, caused by using `highmem=on` with `-accel hvf`on an M1 mac. See below for more info: https://github.com/utmapp/UTM/issues/3946

**Output of `podman version`:**
7.1.1

```
Client:       Podman Engine
Version:      4.1.1
API Version:  4.1.1
Go Version:   go1.18.3
Built:        Wed Jun 15 05:12:46 2022
OS/Arch:      darwin/arm64

Server:       Podman Engine
Version:      4.1.0
API Version:  4.1.0
Go Version:   go1.18
Built:        Sat May  7 01:16:38 2022
OS/Arch:      linux/arm64
```

**Output of `podman info --debug`:**

```
host:
  arch: arm64
  buildahVersion: 1.26.1
  cgroupControllers:
  - cpu
  - io
  - memory
  - pids
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.1.0-2.fc36.aarch64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.0, commit: '
  cpuUtilization:
    idlePercent: 97.44
    systemPercent: 1.46
    userPercent: 1.09
  cpus: 1
  distribution:
    distribution: fedora
    variant: coreos
    version: "36"
  eventLogger: journald
  hostname: localhost.localdomain
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 1000000
    uidmap:
    - container_id: 0
      host_id: 504
      size: 1
    - container_id: 1
      host_id: 100000
      size: 1000000
  kernel: 5.17.12-300.fc36.aarch64
  linkmode: dynamic
  logDriver: journald
  memFree: 1322921984
  memTotal: 2053279744
  networkBackend: netavark
  ociRuntime:
    name: crun
    package: crun-1.4.5-1.fc36.aarch64
    path: /usr/bin/crun
    version: |-
      crun version 1.4.5
      commit: c381048530aa750495cf502ddb7181f2ded5b400
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +YAJL
  os: linux
  remoteSocket:
    exists: true
    path: /run/user/504/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: true
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.2.0-0.2.beta.0.fc36.aarch64
    version: |-
      slirp4netns version 1.2.0-beta.0
      commit: 477db14a24ff1a3de3a705e51ca2c4c1fe3dda64
      libslirp: 4.6.1
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.3
  swapFree: 0
  swapTotal: 0
  uptime: 8m 39.76s
plugins:
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  volume:
  - local
registries:
  search:
  - docker.io
store:
  configFile: /var/home/core/.config/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /var/home/core/.local/share/containers/storage
  graphRootAllocated: 106825756672
  graphRootUsed: 2297999360
  graphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 0
  runRoot: /run/user/504/containers
  volumePath: /var/home/core/.local/share/containers/storage/volumes
version:
  APIVersion: 4.1.0
  Built: 1651853798
  BuiltTime: Sat May  7 01:16:38 2022
  GitCommit: ""
  GoVersion: go1.18
  Os: linux
  OsArch: linux/arm64
  Version: 4.1.0
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
$ brew info podman
podman: stable 4.1.1 (bottled), HEAD
Tool for managing OCI containers and pods
https://podman.io/
/opt/homebrew/Cellar/podman/4.1.1 (175 files, 77.7MB) *
  Poured from bottle on 2022-06-16 at 10:32:10
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/podman.rb
License: Apache-2.0
==> Dependencies
Build: go ✔, go-md2man ✔
Required: qemu ✔
==> Options
--HEAD
    Install HEAD version
==> Caveats
zsh completions have been installed to:
  /opt/homebrew/share/zsh/site-functions
==> Analytics
install: 18,206 (30 days), 59,671 (90 days), 157,270 (365 days)
install-on-request: 18,077 (30 days), 59,510 (90 days), 157,087 (365 days)
build-error: 1 (30 days)
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide? (https://github.com/containers/podman/blob/main/troubleshooting.md)**

Yes

**Additional environment details (AWS, VirtualBox, physical, etc.):**

N/A

**Comment by akirilov** [comment]
Looking more closely at 8d3e6577ae4b5c6a6ed25ff5c332df042094e415, it seems reverting it would cause issues with certain systems as well, based on the commit message. My repro was on qemu `7.0.0`, in case that is different than the original author's version.

**Comment by akirilov** [comment]
Upon further investigation, seems to be an issue with qemu 7.0.0 on OS X 12.3 as discussed in the linked qemu issue.

Was not able to repro on OS X 12.4

**Comment by rhatdan** [comment]
This seems to be an issue between OS X and Qemu, not something Podman can fix.  Problem is fixed as everyone updatest to 12.4.  I am going to close this issue, but conversation can continue here.

**Comment by akirilov** [comment]
Agreed with that resolution!
